### PR TITLE
Fixes #1658: NullPointerException at HttpClientBuilder

### LINKED ISSUE #1658: java.lang.NullPointerException:    at openfoodfacts.github.scrachx.openfood.utils.Utils.HttpClientBuilder
```
30 minutes ago on app version 52
Xiaomi Redmi Note 3 (hennessy), 2048MB RAM, Android 5.0
Report 1 of 1
java.lang.RuntimeException: 
  at android.app.ActivityThread.performStopActivityInner (ActivityThread.java)
  at android.app.ActivityThread.handleStopActivity (ActivityThread.java)
  at android.app.ActivityThread.access$1200 (ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java)
  at android.os.Handler.dispatchMessage (Handler.java)
  at android.os.Looper.loop (Looper.java)
  at android.app.ActivityThread.main (ActivityThread.java)
  at java.lang.reflect.Method.invoke (Method.java)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java)
Caused by: java.lang.NullPointerException: 
  at openfoodfacts.github.scrachx.openfood.utils.Utils.HttpClientBuilder (Utils.java)
  or                     .apply (Utils.java)
  or                     .bold (Utils.java)
  or                     .compressImage (Utils.java)
  or                     .convertKjToKcal (Utils.java)
  or                     .decodeFile (Utils.java)
  or                     .getAppDaoSession (Utils.java)
  or                     .getBitmapFromDrawable (Utils.java)
  or                     .getClickableText (Utils.java)
  or                     .hideKeyboard (Utils.java)
  or                     .isApplicationInstalled (Utils.java)
  or                     .lambda$scan$0$Utils (Utils.java)
  or                     .openTags (Utils.java)
  at openfoodfacts.github.scrachx.openfood.fragments.SaveOfflineSummaryFragment.getProduct (SaveOfflineSummaryFragment.java)
  at openfoodfacts.github.scrachx.openfood.views.SaveProductOfflineActivity.lambda$onRequestPermissionsResult$5$SaveProductOfflineActivity (SaveProductOfflineActivity.java)
  or                     .saveProduct (SaveProductOfflineActivity.java)
  at openfoodfacts.github.scrachx.openfood.views.SaveProductOfflineActivity.onStop (SaveProductOfflineActivity.java)
  at android.app.Instrumentation.callActivityOnStop (Instrumentation.java)
  at android.app.Activity.performStop (Activity.java)
  at android.app.ActivityThread.performStopActivityInner (ActivityThread.java)
  at android.app.ActivityThread.handleStopActivity (ActivityThread.java)
  at android.app.ActivityThread.access$1200 (ActivityThread.java)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java)
  at android.os.Handler.dispatchMessage (Handler.java)
  at android.os.Looper.loop (Looper.java)
  at android.app.ActivityThread.main (ActivityThread.java)
  at java.lang.reflect.Method.invoke (Method.java)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java)
```

### PR DESCRIPTION
**Pull request by janmiga**
## Description

Added a null check on the passed activity in Utils.hideKeyboard to fix a crash in SaveOfflineSummaryFragment

## Related issues and discussion
#1658 
 
 ## Checklist
 
Make sure you've done all the following (You can delete the checklist before submitting)
 
 - [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.
 - [x] Code is well documented
 - [x] Include unit tests for new functionality
 - [x] All user-visible strings are made translatable
 - [ ] Code passes Travis builds in your branch
 - [x] If you have multiple commits please combine them into one commit by squashing them.
 - [x] Read and understood the contribution guidelines .

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import android.Manifest;
 import android.annotation.TargetApi;
 import android.app.Activity;
+import android.app.Fragment;
 import android.content.Context;
```
Comment (huzaifaiftikhar) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java]: Isn't this an unused import statement?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java [hunk 2]
```diff
@@ -132,14 +133,20 @@ public static CharSequence bold(CharSequence... content) {
     }
 
     public static void hideKeyboard(Activity activity) {
-        View view = activity.getCurrentFocus();
+        if (activity == null)
+            return;
```
Comment (huzaifaiftikhar) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java]: I guess just adding these two lines to the top of the method will solve the issue. The other additions are logically identical to the previous version.

Comment (janmiga) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java]: exactly, do you want me to undo the other changes or leave them (as they improve readability)?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java (hunk 1 above)
Comment (janmiga) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java]: You are right, I removed that line

**Comment by janmiga** [comment]
done

**Comment by Karljoones** [comment]
Merged, thanks @janmiga

**Pull request merged**